haystack/components/generators/chat/openai.py

```diff
@@ -280,9 +301,10 @@ def run(
             tools=tools,
             tools_strict=tools_strict,
         )
-        chat_completion: Union[Stream[ChatCompletionChunk], ChatCompletion] = self.client.chat.completions.create(
-            **api_args
-        )
+        if "stream" in api_args.keys():
+            chat_completion = self.client.chat.completions.create(**api_args)
```

Comment:
I find this hard to understand. Something similar to this would work?

We could always return a dictionary with the same fields from `_prepare_api_call`.

```python
        if api_args.get("response_format"):
            # We cannot pass stream param to chat.completions.parse endpoint
            api_args.pop("stream", None)
            chat_completion = self.client.chat.completions.parse(**api_args)
      else:
            ...
```

(I might be wrong)